tests/integration/run-prettier.js

```diff
@@ -40,20 +40,15 @@ async function run(dir, args, options) {
 
   jest.spyOn(Date, "now").mockImplementation(() => 0);
 
-  const orig_writeFile = fs.promises.writeFile;
   const write = [];
 
   jest
     .spyOn(fs.promises, "writeFile")
-    .mockImplementation(async function(filename, content) {
-      if (options.writeFileCallback) {
-        const spied_writeFile = fs.promises.writeFile;
-        fs.promises.writeFile = orig_writeFile;
-        try {
-          await options.writeFileCallback.apply(options, arguments);
-        } finally {
-          fs.promises.writeFile = spied_writeFile;
-        }
+    // eslint-disable-next-line require-await
+    .mockImplementation(async (filename, content) => {
+      const error = options.mockWriteFileErrors?.[filename];
```

Comment:
Is the filename an absolute path? `path.relative()`?

Reply:
as is?

Reply:
I think this is fine.